doc/design/s3-backup.md

```diff
@@ -0,0 +1,84 @@
+
+## Limitations and Considerations
+
+- **Delete marker replication only works for soft deletes**: If an object version is explicitly deleted (by version ID, i.e., a “hard delete”), this deletion is NOT replicated. This does not affect DANDI, as explicit version deletion never happens in normal operations; in fact, it is explicitly disallowed via bucket policy and is deferred to a lifecycle policy (see Trailing Delete design doc for more information).
+- **Replication is eventually consistent**: AWS does not provide any guarantees about replication speed (the time between an object finishing upload into the primary bucket and when it is available in the backup bucket). They offer a feature called "Replication Time Control" that guarantees a 15-minute SLA, but it's unclear if this is needed for DANDI. It costs an additional $0.015 per GB of data transferred (see [AWS docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-time-control.html)).
+- **Existing objects are not automatically replicated**: Replication only applies to objects created or updated after replication is enabled. For existing objects, a one-time sync operation will be required.
+- **Glacier Deep Archive has a minimum retention period of 180 days:** the backup bucket will use the Glacier Deep Archive storage class, as it costs substantially less than the Standard storage class ($0.00099/GB vs ~$0.20/GB, respectively), but has a minimum cost of 180 days of storage per object. That is, any retention period of less than 180 days will incur the same cost as if it were 180 days. This likely impacts the cost analysis of this design.
+- **Glacier Deep Archive requires 12-48 hours to retrieve data:** the exact amount of time depends on the retrieval mechanism (see [AWS docs](https://aws.amazon.com/s3/storage-classes/glacier/)), but either way, instantaneous recovery is not possible. However, this latency is acceptable for purposes of catastrophic data recovery; it also indicates why the trailing delete mechanism continues to be valuable (it offers a much faster "data recovery" in the cases where it can be used).
+
+## Cost
+
+**Below are the additional costs introduced by this backup feature** for a 1 PB primary bucket (assuming both the primary bucket and backup bucket are in us-east-2). All of this information was gathered from the "Storage & requests", "Data transfer", and "Replication" tabs on [https://aws.amazon.com/s3/pricing/](https://aws.amazon.com/s3/pricing/).
+
+**Storage Costs** (backup bucket in us-east-2):
+
+- Glacier Deep Archive storage: ~$0.00099/GB/month
+    - 1 PB = 1,024 TB × $0.99/TB = **$1,014/month**
+
+**Data Transfer Costs**:
+
+- Same-region data transfer between S3 buckets is free
```

Comment:
The definition on line https://github.com/dandi/dandi-archive/pull/2627/files#diff-9a9b9bb814eeea229ca42a1f536abf895736f6aeb248def55cc122469f619d0fR6 indicates the backup will live in a separate region though?

Reply:
Ah thanks, I missed that - we originally considered a bucket in a separate region, but settled on same-region. I pushed a fix